### ISSUE TITLE
Custom dictionary / generators / card decks / oracles + ideas

### ISSUE THREAD

**Issue by RodWarrior** [alexkurowski/solo-toolkit#4]
Hello! I love this plugin! Thanks for your hard work. Also, I have a few ideas i'd like to share. I'm from Brazil, so i speak portuguese mainly - sorry for any errors in my english. 

1) Customization for default generators

First, I would like to have my own custom look ups for all the functions basically. It would be nice if they were open to edit like the custom table one. As I am brazilian, it would be rather helpful if I could have the generators in my own language. So i would love the option to change the text in all the generators. Also, would be fun to be able to add my own stuff - so i could make personalized decks, tables, dictionaries to use on the plugin. For example, a full tarot deck with all 78 cards with small descriptions. Or a custom deck for other uses or specific interpretations. Maybe an oracle with my own text (Instead of "Yes, but.." - "Sim, mas.." or to add even more options). That would be nice to add your own classes / skills / aspects or ocupations, or at least translate the ones that are there.

TLDR: Basically, the same idea for the custom table templates, but applied to every other generator - decks, oracles and dictionaries. Also, the ability to change the default text.

1.1) The ability to show or hide default generatos

If we can't change the default ones, would be nice to have the templates to use as basis for our custom ones, and the ability to hide the defaults from view.

1.2) Custom Oracles and Categories

It would be nice being able to add new options to oracles, maybe also categorized like tables are right now. I know probably we can do this with tables right now, but if they function the same, it would be nice to have the option to organize like in tables - so we could have narrative, skill checks, actions or various subjects. Example for this would be genres (sci-fi, fantasy, etc..), context (combat actions, conversation actions) or even quality (Evil, Good, Friendly, Hostile).

1.3) Custom Dictionaries

It would be lovely being able to set up my own dictionaries, so that it could be a "generic" categories used by all others tables. I'm not sure if custom tables already function like this, need to test it a bit more.

1.4) Cross-table reference

It would be fun to being able to reference other tables or their sections somehow. This seems like a huge work, so i don't know if it's possible. Doing something like {tablename.keyword} to reuse some other table keywords and having to update in only one file. This could also be done by using the custom dictionaries as global lists that all tables could "call", reducing the need to update all the lists that contain those same elements.

2) Images for decks

It would be really cool to be able to put image links on cards, so that we could have multiple decks with different images from each other. For example, have a Ryder and Crowley Tarot - or even custom ones. Maybe it could be a folder (/Decks) where we could put images in subfolder (/Decks/Crowley) where images could be the name of cards or numbered from 1 to i. 

2.1) Descriptions for cards in decks

I saw that in the source you have a small descriptor for each card. That could be really useful if it could be acessed and customized for new and default decks. Maybe we could use a deck for actions or events, in a way that it wouldn't repeat until shuffle. That would be really nice to flesh out some options or clarify different interpretations for tarot decks for example (Like having one focusing on narrative interpretations, other focusing on character qualities, other for events, etc...)

Sorry for the big text, these are some ideas that I would like to share about this lovely plugin. I found it really cool. Mostly it could be done already with custom tables (except the deck parts) so i tought i could share here. I don't know how hard each of these would be, but it's some of the things I was thinking while learning how to use this. Thanks for you work, really liked this plugin! I don't know hou much of all of this would be doable or possible, but felt the urge to share to at least collaborate a bit with your awesome plugin.

**Comment by alexkurowski** [comment]
Great suggestions all around!

I've just release version 0.3.0, where I opted for decks to be all images. I nearly forgot that I had a collection of GMA cards for VTT that I was eager to use 😅 
There's now an option for a "Decks" folder, where you can add subfolders with images of your cards.

Also, you can refer to other custom tables with property keywords `{note_name}`, or `{folder_name/note_name/category_name}` for more control. I hope you find it useful!

As for translations, I haven't really thought about it. The yes/no and/but oracle should be simple to translate, I'll look into it later. Though, other dictionaries are pretty big and convoluted, so it might be harder to translate (and different languages have different rules). Maybe I could add an "Export" button in settings that would create all the default tables that you could run through AI or edit however you like.

**Comment by RodWarrior** [comment]
Thank you! With the custom table references i can basically do everything now, having tables as dictionaries with no problem. Oracles would be cool to have varied types, but i think it essentialy works as a table too? So no problem there also.

I'll test the deck function later, but it looks awesome!

You did an amazing and super fast job, i'm really grateful! Awesome work!
